common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java

```diff
@@ -111,24 +110,14 @@ public static PooledByteBufAllocator createPooledByteBufAllocator(
     }
     return new PooledByteBufAllocator(
       allowDirectBufs && PlatformDependent.directBufferPreferred(),
-      Math.min(getPrivateStaticField("DEFAULT_NUM_HEAP_ARENA"), numCores),
-      Math.min(getPrivateStaticField("DEFAULT_NUM_DIRECT_ARENA"), allowDirectBufs ? numCores : 0),
-      getPrivateStaticField("DEFAULT_PAGE_SIZE"),
-      getPrivateStaticField("DEFAULT_MAX_ORDER"),
-      allowCache ? getPrivateStaticField("DEFAULT_TINY_CACHE_SIZE") : 0,
-      allowCache ? getPrivateStaticField("DEFAULT_SMALL_CACHE_SIZE") : 0,
-      allowCache ? getPrivateStaticField("DEFAULT_NORMAL_CACHE_SIZE") : 0
+      Math.min(PooledByteBufAllocator.defaultNumHeapArena(), numCores),
```

Comment:
I had shied away from touching the Netty warnings in the past because of some uncertainty about different versions coming in at runtime, but, I think that's not a valid concern now. I think this is good to fix up. Anything that's in Netty 4.1+ is fair game.